libs/shared/src/hooks/use-indexer-balance-query/use-indexer-balance-query.tsx

```diff
@@ -50,61 +14,140 @@ export interface IndexerBalances {
+    method: 'updates_v2',
+    params: [addresses, date.toISOString(), currency],
+  };
+}
+
+// Parse the balance from the indexer response
+function safeParseBalance(
+  balances: IndexerBalance,
+  address: string,
+  chainId: number,
+): number {
+  const balance = balances.find(([addr, chain]) => {
+    return addr === address && chain === chainId;
+  });
+
+  if (!balance) {
+    return 0;
+  }
+
+  return Number.parseFloat(formatUnits(BigInt(balance[2]), 18));
+}
+
+// Map the balances from the indexer response to the IndexerBalances type
 function mapBalances(
-  balancesResponse: IndexerBalancesResponse,
+  balancesResponse: IndexerV2UpdatesResponse,
   address: string,
+  chainId: number,
 ): IndexerBalances {
   return {
-    available: parseResponseNumber(balancesResponse.available[address]),
-    availableForStake: parseResponseNumber(
-      balancesResponse.available_for_stake[address],
+    available: safeParseBalance(balancesResponse.available, address, chainId),
+    availableForStake: safeParseBalance(
+      balancesResponse.available_for_stake,
+      address,
+      chainId,
+    ),
+    balance: safeParseBalance(balancesResponse.balance, address, chainId),
+    locked: safeParseBalance(balancesResponse.locked, address, chainId),
+    staked: safeParseBalance(balancesResponse.staked, address, chainId),
+    stakedFree: safeParseBalance(
+      balancesResponse.staked_free,
+      address,
+      chainId,
     ),
-    balance: parseResponseNumber(balancesResponse.balance[address]),
-    locked: parseResponseNumber(balancesResponse.locked[address]),
-    staked: parseResponseNumber(balancesResponse.staked[address]),
-    stakedFree: parseResponseNumber(balancesResponse.staked_free[address]),
-    stakedLocked: parseResponseNumber(balancesResponse.staked_locked[address]),
-    total: parseResponseNumber(balancesResponse.total[address]),
-    vested: parseResponseNumber(balancesResponse.vested[address]),
-    daoLocked: parseResponseNumber(balancesResponse.dao_locked[address]),
+    stakedLocked: safeParseBalance(
+      balancesResponse.staked_locked,
+      address,
+      chainId,
+    ),
+    total: safeParseBalance(balancesResponse.total, address, chainId),
+    vested: safeParseBalance(balancesResponse.vested, address, chainId),
+    daoLocked: safeParseBalance(balancesResponse.dao_locked, address, chainId),
+    unbonding: safeParseBalance(balancesResponse.unbounding, address, chainId),
+    rewards: safeParseBalance(balancesResponse.rewards, address, chainId),
   };
 }
 
+// Fetch the balances from the indexer
 export async function indexerBalancesFetcher(
   chainId: number,
   address?: string,
 ) {
   if (!address) {
     return null;
   }
-  const isTestedge = chainId === haqqTestedge2.id;
+
+  const addresses = { [chainId]: [address] };
+
   const requestUrl = new URL(
-    isTestedge ? TESTEDGE_INDEXER_ENDPOINT : INDEXER_ENDPOINT,
+    'https://backend.wallet.production.haqq.network/api/all_networks/jsonrpc',
```

Comment:
Always production?

Reply:
Should we use the dev backend for testedge?
In what scenarios should we prefer the dev backend over production?

Reply:
in case of testing on testedge2

maybe it can cause some problems for QA (Ivan). He will see prod-values on non-prod environment